src/com/google/guiceberry/junit4/GuiceBerryTestRule.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.guiceberry.junit4;
+
+import com.google.guiceberry.DefaultEnvSelector;
+import com.google.guiceberry.GuiceBerry;
+import com.google.guiceberry.GuiceBerryEnvSelector;
+import com.google.guiceberry.TestDescription;
+import com.google.inject.Module;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * {@link GuiceBerry} adapter for JUnit4 tests, implementing {@TestRule}.
+ *
+ * <p>Please do not use this unless there is a clear reason that {@link GuiceBerryRule} won't
+ * work, for example if you need the rule to be part of a {@link RuleChain}.
+ *
+ * @author David M. Hull
+ */
+public final class GuiceBerryTestRule extends GuiceBerryBaseRule implements TestRule {
+  private final Object target;
+
+  /**
+   * Main constructor.
+   *
+   * @param target Must be the test object.  The usual construct is
```

Comment:
The comment about "the usual construct" as written is more appropriate as a method comment rather than a comment to this param. I would do that, and leave here:

@param target the test object. It's always {@code this}.